File: spec/unit/frame_spec.rb
require "frame"

describe Frame do
  it "expects to know about roll it is on" do
    frame = Frame.new
    expect(frame.rolls).to eq 0
  end

  it "knows if frame was a spare" do
    frame = Frame.new
    expect(frame.is_spare?).to eq false
    frame.add_roll(5)
    frame.add_roll(5)
    expect(frame.is_spare?).to eq true
    expect(frame.is_strike?).to eq false
  end

  it "doesn't allow the player to input more pins felled than available" do
    frame = Frame.new
    frame.add_roll(5)
    expect { frame.add_roll(6) }.to raise_error "You cannot down more pins than available"
  end

  it "doesn't let you enter more than two rolls" do
    frame = Frame.new
    frame.add_roll(5)
    frame.add_roll(4)
    expect { frame.add_roll(1) }.to raise_error "You have already finished this frame"
  end

  it "doesn't allow you to enter any more rolls after a strike" do
    frame = Frame.new
    frame.add_roll(10)
    expect { frame.add_roll(1) }.to raise_error "You have already finished this frame"
  end

  it "doesn't allow you to enter any more rolls after a spare" do
    frame = Frame.new
    frame.add_roll(5)
    frame.add_roll(5)
    expect { frame.add_roll(1) }.to raise_error "You have already finished this frame"
  end
end

File: jasmine-standalone-3.6.0/spec/FrameSpec.js
describe("Frame", () => {
  let frame

  it('checks for a strike', () => {
    frame = new Frame([10])
    expect(frame.isStrike()).toEqual(true)
  })

  it('checks for a spare', () => {
    frame = new Frame([3, 7])
    expect(frame.isSpare()).toEqual(true)
  })

  it('returns the roll score', () => {
    frame = new Frame([3, 7])
    expect(frame.score()).toEqual(10)
  })

  it('returns the first roll', () => {
    frame = new Frame([5, 2])
    expect(frame.first()).toEqual(5)
  })

  it('scores the last roll', () => {
    frame = new Frame([10, 10, 10])
    expect(frame.tenth()).toEqual(30)
  })
})
